### PR TITLE
(testing): Add unit tests for `pkg/cache/components`

### LINKED ISSUE #7: Write unit tests for cache components
## Description
We need to write unit tests for all the components that make up a cache found in https://github.com/everettraven/telescopia/blob/main/pkg/cache/components.go

If needed, some refactoring can be done to make the code easier to test, read, understand, and maintain.

### PR DESCRIPTION
**Pull request by everettraven**
## Description
- Add unit tests for `pkg/cache/components` package
- Update `pkg/cache/components/scope_informer.go` due to a nil reference of the `ListOptions.LabelSelector` causing the informer to panic when performing a `List()` without setting the `ListOptions.LabelSelector` field.
    - Now by default a `LabelSelector` returned by `labels.Everything()` will be used if `ListOptions.LabelSelector` is `nil`
- Add a `Makefile` to help simplify the process of running tests, verification checks, etc.
- Add GitHub Actions to run unit tests and verification checks on pushes to `main` and PRs

## Motivation
- resolves #7

### CODE REVIEW

File: pkg/cache/components/scope_informer_test.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+			})
+
+			When("Removing the Dependent", func() {
+				BeforeEach(func() {
+					si.RemoveDependent(dependent)
+				})
+
+				It("Should not have the dependent", func() {
+					Expect(si.HasDependent(dependent)).Should(BeFalse())
+				})
```
Comment (jmrodri) [inline pkg/cache/components/scope_informer_test.go]: Why have the `BeforeEach` here? I probably would've just put them together:

```go
It("Should not have the dependent", func() {
    si.RemoveDependent(dependent)
    Expect(si.HasDependent(dependent)).Should(BeFalse())
})
```

File: pkg/cache/components/scope_informer_test.go [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+			})
+		})
+
+		When("Calling AddEventHandler()", func() {
+			It("Should be added to the ScopeInformer.informer if it hasn't been before", func() {
```
Comment (jmrodri) [inline pkg/cache/components/scope_informer_test.go]: What should be added? Maybe `It("Should add THE THING to the ScopeInformer...` where THE THING is named for what it is we're adding :)

File: pkg/cache/components/component_suite_test.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	// cfg is defined in this file globally.
+	cfg, err = testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	k8sClient, err = kubernetes.NewForConfig(cfg)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient).NotTo(BeNil())
+
+	// initialize a set of namespaces and pods
```
Comment (jmrodri) [inline pkg/cache/components/component_suite_test.go]: I would actually summarize the code.
```suggestion
	// initialize 2 namespaces with 5 pods
```

I know that risks getting out of date, but I can actually SKIM the code after reading that comment.

**Review by jmrodri — CHANGES_REQUESTED** [review]
I have mixed feelings on the `When` statements. It seems to work it's not wrong. I am biased towards `Describe` statements with just the specific method in them.

Keep them for now. I would make sure to describe in your `It` statements WHAT it is being added.

File: pkg/cache/components/scope_informer_test.go (hunk 1 above)
Comment (everettraven) [reply pkg/cache/components/scope_informer_test.go]: Good catch! I placed that `BeforeEach` there when I was first writing that block thinking there would be more than one test case and didn't clean that up :)

Should be cleaned up as of https://github.com/everettraven/telescopia/pull/14/commits/da1f64c076a3c3f721be56e3ac779ca229808298

File: pkg/cache/components/scope_informer_test.go (hunk 2 above)
Comment (everettraven) [reply pkg/cache/components/scope_informer_test.go]: Good point! I think I did this in later tests...

Should be fixed as of https://github.com/everettraven/telescopia/pull/14/commits/da1f64c076a3c3f721be56e3ac779ca229808298

File: pkg/cache/components/component_suite_test.go (hunk 3 above)
Comment (everettraven) [reply pkg/cache/components/component_suite_test.go]: Makes sense. Added a comment in https://github.com/everettraven/telescopia/pull/14/commits/da1f64c076a3c3f721be56e3ac779ca229808298 that should resolve this

**Review by grokspawn — APPROVED** [review]

**Comment by grokspawn** [comment]
I'd still like to see us move to CI using the go-version-file (https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file) at least on all new stuff, but it doesn't have to block this.

**Comment by everettraven** [comment]
> I'd still like to see us move to CI using the go-version-file (https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file) at least on all new stuff, but it doesn't have to block this.

@grokspawn I totally forgot about this. Since it is small I'll go ahead and make that change :)

**Pull request merged**